src/OrleansLoggerProvider/LegacyOrleansLoggingFactoryExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using Microsoft.Extensions.Logging;
+using Orleans.Runtime;
+
+namespace Orleans.Extensions.Logging.Legacy
+{
+    public static class LegacyOrleansLoggingFactoryExtensions
+    {
+        /// <summary>
+        /// Add <see cref="LegacyOrleansLoggerProvider"> with event bulking feature to LoggerFactory
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="logConsumers">log consumers which user want to write log events to</param>
+        /// <param name="eventBulkingConfig">config for event bulking feature</param>
+        /// <returns></returns>
+        [Obsolete("Use Microsoft.Extensions.Logging built-in logger providers")]
+        public static ILoggingBuilder AddLegacyOrleansLogging(
+            this ILoggingBuilder builder,
+            List<ILogConsumer> logConsumers,
+            IPEndPoint ipEndPoint = null,
+            EventBulkingConfig eventBulkingConfig = null)
+        {
+            if (builder == null)
+                throw new ArgumentNullException(nameof(builder));
+            builder.AddMesageBulkingLoggerProvider<LegacyOrleansLoggerProvider>(new LegacyOrleansLoggerProvider(logConsumers, ipEndPoint), eventBulkingConfig);
```

Comment:
As a general rule of thumb, if extension method arguments differ, we should provide default values and forward the calls to the one method which does the work. From here it would be easy to not pass in ```severityOverrides``` and call the method below.

Reply:
which general rule of thumb, maybe refer me a link? And I think as a team we should have an agreement on which coding guideline we are following, it provider convenience for the coder to know what style of code he/she should write, instead of discussing it in PRs